### DISCUSSION TITLE
Missmatching versions of electron-log between main and renderer

### DISCUSSION THREAD

**Discussion by idaamit** [megahertz/electron-log#325] [Q&A] [answered]
I have _main_ and _renderer_. 
I was using version 2.2.1 both in _main_ and _renderer_
I needed to upload log files to cloud, so I wanted to use the new ability of ` log.transports.file.archiveLog ` 
I have upgraded _main_ to 4.4.8
I have upgraded the renderer to _4.4.8_ as well, otherwise renderer stopped writing to log.

I need old _main_ (which works with version 2.2.1) to still work, for backward competability.
The old _main_ (which works with version 2.2.1) stoped printing log.
Is there a way , which _renderer_ which uses version 4.4.8 will work with _main_ which uses 2.2.1  ? or vise versa ?

**Comment by megahertz** [comment, marked as answer]
There's a big difference between 4 and 2 under the hood. Theoretically, that's possible to use v2 or main and v4 on renderer, but not vise versa. Some additional configuration may be required for that. I don't know exactly TBH. And please be aware that it's a matter of change. In v4 main and renderer loggers work independently, but in v5 a renderer code will just send all data to the main process.

**Reply by idaamit** [reply]
These are great news! When do you think v5 will be available for use?
In the meanwhile, I'm struggling with having one renderer (which works in v4) working with two different main versions (v2 and v4).
I'm trying to make the renderer load electron-log from the main dependency.
Can you advise how I can accomplish that?

**Reply by megahertz** [reply]
I can't be sure but I think a beta version will be ready in the end of October. If you want to use v2/v4 before that, I think it's better to use a logger in the main process only. In the renderer, you can send logging calls to the main through IPC.
